osgp/protocol-adapter-dlms/osgp-protocol-adapter-dlms/src/test/java/org/opensmartgridplatform/adapter/protocol/dlms/domain/commands/periodicmeterreads/GetPeriodicMeterReadsCommandExecutorTest.java

```diff
@@ -110,43 +164,82 @@ public void testHappy() throws Exception {
 
-        // TODO: mock other calls
-
-        final PeriodicMeterReadsRequestDto request = new PeriodicMeterReadsRequestDto(periodType, new Date(from),
-                new Date(to));
+        final DlmsMeterValueDto meterValue1 = mock(DlmsMeterValueDto.class);
+        final DlmsMeterValueDto meterValue2 = mock(DlmsMeterValueDto.class);
+        final DlmsMeterValueDto meterValue3 = mock(DlmsMeterValueDto.class);
+        final DlmsMeterValueDto meterValue4 = mock(DlmsMeterValueDto.class);
+        when(this.dlmsHelper.getScaledMeterValue(data1, null, "positiveActiveEnergyTariff1")).thenReturn(meterValue1);
+        when(this.dlmsHelper.getScaledMeterValue(data2, null, "positiveActiveEnergyTariff2")).thenReturn(meterValue2);
+        when(this.dlmsHelper.getScaledMeterValue(data3, null, "negativeActiveEnergyTariff1")).thenReturn(meterValue3);
+        when(this.dlmsHelper.getScaledMeterValue(data4, null, "negativeActiveEnergyTariff2")).thenReturn(meterValue4);
 
         // CALL
         final PeriodicMeterReadsResponseDto result = this.executor.execute(this.connectionManager, device, request);
 
-        // VERIFY
-        verify(this.dlmsMessageListener).setDescription(
-                String.format("GetPeriodicMeterReads DAILY from %s until %s, retrieve attribute: {%s,%s,%s}",
-                        new DateTime(from), new DateTime(to), classId, address, id));
+        // VERIFY calls to mocks
+        verify(this.dlmsMessageListener).setDescription(String.format(
+                "GetPeriodicMeterReads DAILY from %s until %s, retrieve attribute: {%s,%s,%s}",
+                new DateTime(this.from), new DateTime(this.to), dlmsProfile.getClassId(), dlmsProfile.getObisCode(),
+                dlmsProfile.getDefaultAttributeId()));
 
-        verify(this.dlmsHelper).validateBufferedDateTime(same(bufferedDateTime), same(cosemDateTime),
+        verify(this.dlmsHelper, times(2)).validateBufferedDateTime(same(bufferedDateTime),
                 argThat(new DateTimeMatcher(from)), argThat(new DateTimeMatcher(to)));
 
+        verify(this.dlmsObjectConfigService).findDlmsObject(any(Protocol.class),
+                any(DlmsObjectType.class), any(Medium.class));
+
+        // ASSERT - the result should contain 2 values
         final List<PeriodicMeterReadsResponseItemDto> periodicMeterReads = result.getPeriodicMeterReads();
-        // TODO: assert contents of result
+
+        assertThat(periodicMeterReads.size()).isEqualTo(2);
+
+        periodicMeterReads.forEach(p -> {
+            assertThat(p.getLogTime()).isNotNull();
+            assertThat(p.getActiveEnergyImportTariffOne()).isEqualTo(meterValue1);
+            assertThat(p.getActiveEnergyImportTariffTwo()).isEqualTo(meterValue2);
+            assertThat(p.getActiveEnergyExportTariffOne()).isEqualTo(meterValue3);
+            assertThat(p.getActiveEnergyExportTariffTwo()).isEqualTo(meterValue4);
+        });
+
+    }
```

Comment:
The method `testHappy` is very long. I suggest extracting setup logic as methods to make it more readable.

Reply:
could be done in another story.